collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -2837,8 +2842,28 @@ protected void onStop() {
         super.onStop();
     }
 
-    private void sendSavedBroadcast() {
-        sendBroadcast(new Intent("org.odk.collect.android.FormSaved"));
+    /**
+     * Enqueue an autosend job if the form that was just finalized needs to be autosent. That can
+     * happen either because the blank form specifies that it should be autosent or because app-
+     * wide-autosend is enabled.
+     *
+     * TODO: if the user changes autosend settings, should an autosend job immediately be enqueued?
```

Comment:
I was wondering this too.  I'd like to know more about the thinking behind why it's better to schedule a job at form finalization time than to check for forms to send at autosend time.

Reply:
I think we definitely want to schedule the job at form finalization time because that form may need to be sent and there may not be an active job. For example, a job could have successfully sent all the forms last time connectivity came on, now connectivity is still available and so we want that one form to be sent right away.

But scheduling a job right when the user turns on autosend also seems like a good idea to me.